calico-vpp-agent/policy/policy_server.go

```diff
@@ -579,12 +674,79 @@ func (s *Server) handleActiveProfileRemove(msg *proto.ActiveProfileRemove, pendi
 }
 
 func (s *Server) handleHostEndpointUpdate(msg *proto.HostEndpointUpdate, pending bool) (err error) {
-	s.log.Infof("Ignoring HostEndpointUpdate")
+	state := s.currentState(pending)
+	id := fromProtoHostEndpointID(msg.Id)
+	hep := fromProtoHostEndpoint(msg.Endpoint, s)
+	if hep.InterfaceName != "" && hep.InterfaceName != "*" {
+		swifindex, err := s.vpp.SearchInterfaceWithTag("main-" + hep.InterfaceName)
+		if err != nil {
+			return err
+		} else if swifindex == types.InvalidID {
+			s.log.Errorf("cannot find host endpoint: interface named %s does not exist", hep.InterfaceName)
+		} else {
+			hep.UplinkSwIfIndexes = append(hep.UplinkSwIfIndexes, swifindex)
+		}
+		hep.TapSwIfIndexes = append(hep.TapSwIfIndexes, s.uplinkToTap[swifindex])
+	} else if hep.InterfaceName == "" && hep.expectedIPs != nil {
+		hep.UplinkSwIfIndexes, err = s.vpp.SearchInterfacesWithAddresses(hep.expectedIPs, "main-")
+		if err != nil {
+			return err
+		}
+		for _, swifindex := range hep.UplinkSwIfIndexes {
+			hep.TapSwIfIndexes = append(hep.TapSwIfIndexes, s.uplinkToTap[swifindex])
+		}
+	} else if hep.InterfaceName == "*" {
+		for uplink, tap := range s.uplinkToTap {
+			hep.UplinkSwIfIndexes = append(hep.UplinkSwIfIndexes, uplink)
+			hep.TapSwIfIndexes = append(hep.TapSwIfIndexes, tap)
+		}
+	}
+	for _, provider := range s.routingServer.ConnectivityServer.Providers {
```

Comment:
This isn't great from an encapsulation perspective, the policy server shouldn't need to know that there are several Providers and a Connectivity server in the routing server. I think it would be a bit better to add a GetAllTunnels method on the routing server. This way, we could switch the ConnectivityServer and the Providers back to private (lowercase) as well.